types/index.d.ts

```diff
@@ -0,0 +1,435 @@
+     * @default true
+     */
+    cellFormula?: boolean;
+
+    /**
+     * Parse rich text and save HTML to the .h field
+     * @default true
+     */
+    cellHTML?: boolean;
+
+    /**
+     * Save number format string to the .z field
+     * @default false
+     */
+    cellNF?: boolean;
+
+    /**
+     * Save style/theme info to the .s field
+     * @default false
+     */
+    cellStyles?: boolean;
+
+    /**
+     * Store dates as type d (default is n)
+     * @default false
+     */
+    cellDates?: boolean;
+
+    /**
+     * Create cell objects for stub cells
+     * @default false
+     */
+    sheetStubs?: boolean;
+
+    /**
+     * If >0, read the first sheetRows rows
+     * @default 0
+     */
+    sheetRows?: number;
+
+    /**
+     * If true, parse calculation chains
+     * @default false
+     */
+    bookDeps?: boolean;
+
+    /**
+     * If true, add raw files to book object
+     * @default false
+     */
+    bookFiles?: boolean;
+
+    /**
+     * If true, only parse enough to get book metadata
+     * @default false
+     */
+    bookProps?: boolean;
+
+    /**
+     * If true, only parse enough to get the sheet names
+     * @default false
+     */
+    bookSheets?: boolean;
+
+    /**
+     * If true, expose vbaProject.bin to vbaraw field
+     * @default false
+     */
+    bookVBA?: boolean;
+
+    /**
+     * If defined and file is encrypted, use password
+     * @default ''
+     */
+    password?: string;
+}
+
+export interface WritingOptions {
+    /**
+     * Output data encoding
+     */
+    type?: 'base64' | 'binary' | 'buffer' | 'file';
+
+    /**
+     * Store dates as type d (default is n)
+     * @default false
+     */
+    cellDates?: boolean;
+
+    /**
+     * Generate Shared String Table
+     * @default false
+     */
+    bookSST?: boolean;
+
+    /**
+     * Type of Workbook
+     * @default 'xlsx'
+     */
+    bookType?: 'xlsx' | 'xlsm' | 'xlsb' | 'ods' | 'biff2' | 'fods' | 'csv';
+
+    /**
+     * Name of Worksheet for single-sheet formats
+     * @default ''
+     */
+    sheet?: string;
+
+    /**
+     * Use ZIP compression for ZIP-based formats
+     * @default false
+     */
+    compression?: boolean;
+}
+
+export interface WorkBook {
+    /**
+     * A dictionary of the worksheets in the workbook.
+     * Use SheetNames to reference these.
+     */
+    Sheets: { [sheet: string]: WorkSheet };
+
+    /**
+     * ordered list of the sheet names in the workbook
+     */
+    SheetNames: string[];
+
+    /**
+     * an object storing the standard properties. wb.Custprops stores custom properties.
+     * Since the XLS standard properties deviate from the XLSX standard, XLS parsing stores core properties in both places.
+     */
+    Props: Properties;
+}
+
+export interface ColInfo {
+    /**
+     * Excel's "Max Digit Width" unit, always integral
+     */
+    MDW?: number;
+    /**
+     * width in Excel's "Max Digit Width", width*256 is integral
+     */
+    width: number;
```

Comment:
@DMan577 this was a mistake from the flow definitions, width is optional and we'll correct it after the merge.